PLAT-10530 Presence service implementation

symphony-youness opened the pull request:
### Ticket
[PLAT-10530](https://perzoinc.atlassian.net/browse/PLAT-10530)

### Description
- implementation
- tests
- added typehint to the ServiceFactory.get_health_service function
- As I understood, the symphony developers documentation is enough for thisservice: https://developers.symphony.com/restapi/reference#get-presence

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [x] Updated the documentation in [docs folder](../docs)

symphony-youness committed:
PLAT-10530 Presence service implementation
- implementation
- partial tests
- added typehint to the ServiceFactory.get_health_service function

symphony-youness committed:
PLAT-10530 Completed Presence service tests
- put test payloads in one json file

symphony-youness committed:
PLAT-10530 Presence service BDK entry point

symphony-youri commented on symphony/bdk/core/service/presence/presence_service.py:
```diff
@@ -0,0 +1,176 @@
+            session_token=await self._auth_session.session_token, feed_id=feed_id)
+        return string_id.id
+
+    async def set_user_presence(self, user_id, status, soft):
```
can't we type the status with the enum here? (and soft too, and user_id)

symphony-youri commented on symphony/bdk/core/service/presence/presence_service.py:
```diff
@@ -0,0 +1,176 @@
+            session_token=await self._auth_session.session_token)
+        return string_id.id
+
+    async def read_presence_feed(self, feed_id):
```
can we add type hints to this method and the one after

symphony-youness replied:
added arguments and returns type hints to all functions

symphony-youri commented on symphony/bdk/core/service/presence/presence_service.py:
```diff
@@ -0,0 +1,176 @@
+                                                              presence=presence_status,
+                                                              soft=soft)
+
+    async def create_presence_feed(self):
```
we probably did it the same in Java but is the presence feed an OBO service?

thibauult replied:
I do think yes, from https://developers.symphony.com/restapi/reference#obo-enabled-endpoints

symphony-elias replied:
Yes we did the same in Java: https://github.com/finos/symphony-bdk-java/blob/main/symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/presence/OboPresenceService.java

symphony-youri commented on symphony/bdk/core/service_factory.py:
```diff
@@ -119,12 +121,25 @@ def get_datafeed_loop(self) -> AbstractDatafeedLoop:
         :return: a new HealthService instance
         """
-        return HealthService(AgentSystemApi(self._agent_client), SignalsApi(self._agent_client))
+        return HealthService(
```
do we have different formatting config or was it not formatted before?

symphony-youness replied:
It was not formatted before.

symphony-youness committed:
PLAT-10530 updated arguments and return type hints

symphony-youri commented on tests/resources/presence_response/presence_payload.json:
```diff
@@ -0,0 +1,39 @@
+{
```
it becomes more difficult to me to see the exact payload returned by a ws

symphony-youness replied:
For me it's a choice. I thought that centring it like this avoided adding json strings in tests making them bulkier. If you wish to do the same as java, tell me and I will.

Reuse file for multiple tests:
I plan to check if adding scope=module for the fixture makes use of the same json for all tests. (we wouldn't open the file for each test)

symphony-youri replied:
@symphony-elias what's your take on this? (putting all the mocked responses in the same file)

symphony-elias commented on symphony/bdk/core/service/presence/presence_service.py:
```diff
@@ -0,0 +1,176 @@
+    Set Presence <https://developers.symphony.com/restapi/reference#set-presence>`_
+    """
+
+    def _generate_next_value_(name, start, count, last_values):
```
This is fine but you could also use the `name` field to get the enum name as String, like `PresenceStatus.AVAILABLE.name` will return `"AVAILABLE"`.

symphony-youness replied:
sure, I just thought that using value can be more concise. I will change it and remove the overridden method

symphony-elias commented on symphony/bdk/core/service/presence/presence_service.py:
```diff
@@ -0,0 +1,176 @@
+        :return: Presence info of the calling user.
+        `Set Presence <https://developers.symphony.com/restapi/reference#set-presence"`_
+        """
+        presence_status: V2PresenceStatus = V2PresenceStatus(category=status.value)
```
Regarding my comment on PresenceStatus enum, you could use `status.name` instead.

symphony-youness committed:
PLAT-10530 Added documentation
- added smaller (conscise) documentation
- fixed typo in the connection service documentation

symphony-youness committed:
fixup! PLAT-10530 used Enum name access instead of value access

symphony-elias reviewed (approved):
LGTM !
